# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/composite.rs
# Commit: f3b490c076d6
# Full Hash: f3b490c076d623d6ef3bcfd297a0522d9727477c
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-23 21:42:24
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 0c5fb3a8a94d -r f3b490c076d6 gfx/wr/webrender/src/composite.rs
--- a/gfx/wr/webrender/src/composite.rs	Thu Jan 23 15:35:45 2020 +0200
+++ b/gfx/wr/webrender/src/composite.rs	Thu Jan 23 04:37:28 2020 +0000
@@ -94,6 +94,19 @@
     }
 }
 
+impl CompositorConfig {
+    pub fn compositor(&mut self) -> Option<&mut Box<dyn Compositor>> {
+        match self {
+            CompositorConfig::Native { ref mut compositor, .. } => {
+                Some(compositor)
+            }
+            CompositorConfig::Draw { .. } => {
+                None
+            }
+        }
+    }
+}
+
 impl Default for CompositorConfig {
     /// Default compositor config is full present without partial present.
     fn default() -> Self {
@@ -506,6 +519,9 @@
     /// this once when all surface and visual updates are complete, to signal
     /// that the OS composite transaction should be applied.
     fn end_frame(&mut self);
+
+    /// Enable/disable native compositor usage
+    fn enable_native_compositor(&mut self, enable: bool);
 }
 
 /// Return the total area covered by a set of occluders, accounting for